### DISCUSSION TITLE
The https://api.weather.gov/points/... response recently changed 2 properties : updateTie & Icon

### DISCUSSION THREAD

**Discussion by davefxy** [weather-gov/api#733] [Q&A]
The response to the https://api.weather.gov/points/... recently changed 2 properties : updateTime & Icon.
The Icon property changed from a complete Url to a string that no longer a complete Url, it no longer has a website.
The updateTime property changed from a DateTime object to a Date Time string.
So now the [NationalWeatherServiceLibrary ](https://github.com/keanenfarr/NationalWeatherServiceLibrary)  library no longer works.
Am I observing this correctly? And if so where do I find the NWS weather icons?

**Comment by scadergit** [comment]
There are investigating.

**Reply by scadergit** [reply]
They're developing a patch for the icon path. Can you provide more details on the `updateTime` issue? Trying to imagine how a DateTime object could be expressed in JSON.

**Reply by StephenClouse** [reply]
Yeah, I'm confused there as well. That field has always been an ISO8601 string. It still is. Not clear on what's broken, if anything.

**Reply by sullynole** [reply]
@davefxy It appears that 3rd party library is using "updated" and not "updateTime": https://github.com/keanenfarr/NationalWeatherServiceLibrary/blob/6a5e7699ea39512e34b707c4f166364656752ebd/NationalWeatherService/NWS.WeatherDataService/WeatherDataService.cs#L135

Does that explain what you are seeing?  There was an SCN that went out about this expected change: https://www.weather.gov/media/notification/pdf_2023_24/scn24-55_api_v1.13.pdf

**Reply by davefxy** [reply]
I apologize about the updateTime comment. That was me not understanding the code used in dynamic parsing of the time string. I thought I had made a mistake when playing with the code. I assumed there were other changes in the JSON that caused the JSON parsing code to fail. I believe you are correct about the "updated" versus updateTime. I ended up using updateTime to get it to work.
I also thought the 3rd party library was actually from NWS.
I have been using my own code to parse the NWS API replies for a couple of years and was thrilled to find actual code offering an alternative to my own. OOOOPs.
I appreciate your feedback. In the past I found it very difficult to find anyone to communicate with about NWS APIs.
Thank you for responding, Again sorry for the confusion.

**Reply by davefxy** [reply]
I really hope I can continue to use the NWS icon images. I have found them to be so spot on as a weather image. I like them so much better than any other images out there. They can be used to give an instant impression of the weather.

**Reply by davefxy** [reply]
Do you know what is happening with access to the images or new images. In looking for answers about what happened, I discovered the icons have been deprecated in 2022? Is that why the website portion of the icon path may have been dropped ? What is being used by NWS instead. I guess I will find out when they decide what they are going to do.

**Reply by davefxy** [reply]
OMG, I read the SCN https://www.weather.gov/media/notification/pdf_2023_24/scn24-55_api_v1.13.pdf. I wasn't aware this much info was made available to users. I have been reacting to changes in the API responses when my parsing would suddenly blowup. Thanks for pointing this out.

**Reply by scadergit** [reply]
The team was discussing yesterday how the library name is misleading, similar to some app names.

SCNs have always been available on the NWS page, but we're also making significant effort to keep the [API documentation](https://www.weather.gov/documentation/services-web-api) page updated on changes and SCNs specific to the API.

**Comment by davefxy** [comment]
After reviewing my app and the APIs it uses, use MapClick for current weather that includes 5 daily images that are accessible, detailed and short daily forecasts, temp, pop, current hazard, complete current observations including Relh, Dewp, visibility
https://api.weather.gov/gridpoints/BOU/31,61/forecast/hourly for 7 days of hourly info  including Relh, Dewp - no images at the momnent
https://api.weather.gov/gridpoints/BOU/31,61/forecast for 7 days of daily/nightly temp, pop, wind info, forecast text - no images at present

The MapClick response has a radar ID. How can this be used to display radar. There was a time I used this with "https://radar.weather.gov".
I won't continue asking questions after this one.

**Reply by scadergit** [reply]
The icon urls should be fixed shortly.

The Radar app will be updated next week, after which new quick-set URLs will be available for the enhanced maps, for example: /station/KEAX. You can see this in action on the preview for the Radar release: https://preview-radar.weather.gov/station/KEAX. This already works on standard maps: https://radar.weather.gov/station/KEAX/standard. 

Once the API is patched the forecast and hourly forecast endpoint DWML issues, MapClick is on its way out the door, so I would be cautious to use it too much. There will be an SCN for that change.

**Reply by davefxy** [reply]
What will replace MapClick? a series of api requests?

**Reply by scadergit** [reply]
Yes, the existing API service.

**Comment by ktrue** [comment]
Here's what I've observed: The 'icon' entries in the gridpoint JSON have a current format of
```
/icons/land/[day|night]/{icon}[,{pop}[/{icon2}[,{pop2}]?size=medium
```
where `{icon}` is the short name of the icon (no extension) and may include a number like `{icon}{pop}` when it applies to the 12hr period.
The `{icon2}` and `{pop2}` may show if the second 6hr period has a different icon or pop vs. the primary icon, pop.

On forecast.weather.gov forecasts, the icons for display are in `https://forecast.weather.gov/newimages/medium/{icon}.png` or for dual images, use `https://forecast.weather.gov/DualImage.php?i={icon}&ip={pop}&j={icon2}&jp={$pop2}` to dynamically create the dual icon.

The Schema for "icon" entry says 
```
deprecated = true;
A link to an icon representing the forecast summary.
```

So my question is:

Will the API offer an icon or icon code to use that summarizes the conditions during the forecast period?

**Reply by scadergit** [reply]
Yes, before /icon is removed. Otherwise, our own applications that use the API icon path would also break. Likely called something else like key, but tbd, and much anticipated/overdue.

**Reply by vestmon** [reply]
It appears that what is now being returned is a valid endpoint instead of the previous URL. Is this the long term solution for the icon issue?

**Comment by davefxy** [comment]
OMG you guys are soooo col.
Thanks
________________________________

**Comment by roblabs** [comment]
I have updated the Swift package that makes use of one of the changes in [Service Change Notice 24-55][Service Change Notice 24-55].  The [NWS API documentation][NWS API documentation] does mention this change as well.

See https://github.com/WeatherProvider/NationalWeatherService-Swift/pull/17 for the Swift code details.

Thanks to the NWS for both the API and for the discussion site, as I would've missed the change from `updated` to `updateTime`.

[NWS API documentation]: https://www.weather.gov/documentation/services-web-api


[Service Change Notice 24-55]: https://www.weather.gov/media/notification/pdf_2023_24/scn24-55_api_v1.13.pdf
